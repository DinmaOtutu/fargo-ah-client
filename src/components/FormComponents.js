import React from 'react';
import { Form, Label } from 'semantic-ui-react';

const LoginFields = (type, name, onChange, value, placeholder, label, errors = null) => (
  <Form.Field>
    <label htmlFor={type}>
      {label}
      <input
        type={type}
        id={name}
        name={name}
        onChange={onChange}
        value={value}
        placeholder={placeholder}
      />
    </label>
    <Label color="red" pointing>
      {errors}
    </Label>
  </Form.Field>
);

export default LoginFields;
