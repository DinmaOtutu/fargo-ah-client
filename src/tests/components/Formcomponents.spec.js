import React from 'react';
import { shallow } from 'enzyme';
import FormComponents from '../../components/FormComponents';

describe(' LoginModal component rendering', () => {
  const wrapper = shallow(<FormComponents />);
  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
