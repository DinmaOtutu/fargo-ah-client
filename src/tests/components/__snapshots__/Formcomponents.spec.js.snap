// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` LoginModal component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginFields />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          htmlFor={Object {}}
        >
          <input
            id={Object {}}
            name={Object {}}
            type={Object {}}
          />
        </label>,
        <Label
          color="red"
          pointing={true}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <input
              id={Object {}}
              name={Object {}}
              type={Object {}}
            />,
          ],
          "htmlFor": Object {},
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": Object {},
              "name": Object {},
              "onChange": undefined,
              "placeholder": undefined,
              "type": Object {},
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "color": "red",
          "pointing": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            htmlFor={Object {}}
          >
            <input
              id={Object {}}
              name={Object {}}
              type={Object {}}
            />
          </label>,
          <Label
            color="red"
            pointing={true}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <input
                id={Object {}}
                name={Object {}}
                type={Object {}}
              />,
            ],
            "htmlFor": Object {},
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": Object {},
                "name": Object {},
                "onChange": undefined,
                "placeholder": undefined,
                "type": Object {},
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "color": "red",
            "pointing": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
